flake.nix

```diff
@@ -0,0 +1,82 @@
+{
+  description = "mat-chalmers goes flakes";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
```

Comment:
Can we pick a slightly less exciting channel? I think nixos-23.05 is the right amount of exciting right now.

Reply:
Doesn't seem likely. Take `http-client` which has 7.14 on hackage but only 7.13.1 on unstable. I think that as long as we are not updating the lockfile for nix, it will be fine

Reply:
Fair enough, lets try that. A QA tester walks into a bar...